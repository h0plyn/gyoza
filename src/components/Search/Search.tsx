import styled from 'styled-components';
import { SearchProps } from '../../types';

const SearchStyles = styled.div`
  display: flex;
  width: 100%;
  justify-content: flex-end;
  height: 2.1rem;
  & input {
    border-radius: 5px;
    border: 1px solid var(--secondary);
    background-color: var(--card-bg);
    color: var(--highlight);
    padding: 0.5rem;
    font-size: 1.2rem;
    outline: none;
    transition: var(--transition);
    &:focus {
      border: 1px solid var(--headline);
      box-shadow: var(--boxShadow);
    }
  }
`;

export default function Search({ query, handleChange }: SearchProps) {
  return (
    <SearchStyles data-testid="search">
      <label htmlFor="query">
        <input
          type="text"
          name="query"
          value={query}
          onChange={(e) => handleChange(e)}
          placeholder="Coin Name"
          data-testid="input"
        />
      </label>
    </SearchStyles>
  );
}
