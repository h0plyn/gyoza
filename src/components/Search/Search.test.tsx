import { screen, render, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Search from './Search';

describe('<Search />', () => {
  afterEach(() => {
    cleanup();
  });

  const query = 'Ethereum';

  const setup = () => {
    const utils = render(<Search />);
    const input = utils.getByPlaceholderText('Coin Name');
    return { input, ...utils };
  };

  test('it should render the Search component without crashing', () => {
    render(<Search />);
  });

  test('it matches the snapshot of Search', () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders an input field with a descriptive placeholder', () => {
    const { input } = setup();
    expect(input).toBeTruthy();
  });

  test('it calls the handleChange function on change', () => {
    const spy = jest.fn();
    render(<Search handleChange={spy} />);
    const input = screen.getByPlaceholderText('Coin Name');
    fireEvent.change(input, { target: { value: query } });
    expect(spy).toHaveBeenCalledTimes(1);
  });

  test('it takes user entered text', () => {
    const spy = jest.fn();
    render(<Search handleChange={spy} />);
    const input = screen.getByPlaceholderText('Coin Name');
    expect(input.value).toBe('');
    fireEvent.change(input, { target: { value: query } });
    expect(input.value).toBe(query);
  });
});
