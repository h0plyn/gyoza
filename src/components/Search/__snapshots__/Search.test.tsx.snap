// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> it matches the snapshot of Search 1`] = `
<div
  className="sc-bdnxRM hsrMvm"
  data-testid="search"
>
  <label
    htmlFor="query"
  >
    <input
      data-testid="input"
      name="query"
      onChange={[Function]}
      placeholder="Coin Name"
      type="text"
    />
  </label>
</div>
`;
